gx/gx-core/src/main/java/io/graphenee/core/impl/GxMailServiceImpl.java

```diff
@@ -15,39 +15,77 @@
 import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
 import org.springframework.util.StringUtils;
 
+import io.graphenee.core.exception.SendMailFailedException;
+import io.graphenee.core.model.bean.GxMailAttachmentBean;
+import io.graphenee.core.storage.FileStorage;
+import io.graphenee.core.storage.ResolveFailedException;
+
 public class GxMailServiceImpl implements io.graphenee.core.api.GxMailService {
 
 	private JavaMailSender javaMailSender;
 
+	private FileStorage fileStorage;
+
 	@Override
-	public void sendEmail(String subject, String content, String senderEmail, String recipientEmail, String ccEmailList, String bccEmailList) {
-		SimpleMailMessage message = new SimpleMailMessage();
-		message.setSubject(subject);
-		message.setText(content);
-		message.setTo(recipientEmail);
-		message.setFrom(senderEmail);
-		message.setSentDate(new Date());
-		if (!StringUtils.isEmpty(bccEmailList))
-			message.setBcc(bccEmailList);
-		if (!StringUtils.isEmpty(ccEmailList))
-			message.setCc(ccEmailList);
-		getJavaMailSender().send(message);
+	public void sendEmail(String subject, String content, String senderEmail, String recipientEmail, String ccEmailList, String bccEmailList,
+			List<GxMailAttachmentBean> attachmentBeans) throws Exception {
+		try {
+			MimeMessage message = javaMailSender.createMimeMessage();
+			MimeMessageHelper mimeMessage = new MimeMessageHelper(message, true);
+			mimeMessage.setSubject(subject);
+			mimeMessage.setText(content);
+			mimeMessage.setTo(recipientEmail);
+			mimeMessage.setFrom(senderEmail);
+			mimeMessage.setSentDate(new Date());
+			if (!StringUtils.isEmpty(bccEmailList))
+				mimeMessage.setBcc(bccEmailList);
+			if (!StringUtils.isEmpty(ccEmailList))
+				mimeMessage.setCc(ccEmailList);
+			if (attachmentBeans != null)
+				for (GxMailAttachmentBean attachmentBean : attachmentBeans) {
+					String resourcePath = getFileStorage().resourcePath(attachmentBean.getAttachmentFolder(), attachmentBean.getAttachmentFilePath());
+					InputStream inputStream = getFileStorage().resolve(resourcePath);
+					mimeMessage.addAttachment(attachmentBean.getAttachmentName(), new ByteArrayResource(org.apache.poi.util.IOUtils.toByteArray(inputStream)));
+				}
+			javaMailSender.send(message);
+		} catch (ResolveFailedException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			throw new SendMailFailedException(e);
+		}	
 	}
 
 	@Override
-	public void sendEmail(String subject, String content, String senderEmail, String recipientEmail, String ccEmailList) {
-		sendEmail(subject, content, senderEmail, recipientEmail, ccEmailList, null);
+	public void sendEmail(String subject, String content, String senderEmail, String recipientEmail, String ccEmailList) throws Exception {
+		sendEmail(subject, content, senderEmail, recipientEmail, ccEmailList, null, new ArrayList<>());
```

Comment:
Do not pass new ArrayList<>(). create another method called sendEmailWithAttachments() in API.